src/Corax/IndexSearcher/IndexSearcher.cs

```diff
@@ -117,26 +118,26 @@ private Slice EncodeAndApplyAnalyzer(string term, int fieldId)
     //todo maciej: notice this is very inefficient. We need to improve it in future. 
     // Only for KeywordTokenizer
     [SkipLocalsInit]
-    private unsafe ReadOnlySpan<byte> ApplyAnalyzer(ReadOnlySpan<byte> originalTerm, int fieldId)
+    internal unsafe ReadOnlySpan<byte> ApplyAnalyzer(ReadOnlySpan<byte> originalTerm, int fieldId)
     {
-        if (_fieldMapping.Count == 0)
-            return originalTerm;
-
-        if (_fieldMapping.TryGetByFieldId(fieldId, out var binding) == false)
-            return originalTerm;
-        
-        if (binding.FieldIndexingMode == FieldIndexingMode.Exact || binding.Analyzer is null)
+        if (_fieldMapping.TryGetByFieldId(fieldId, out var binding) == false
+            || binding.FieldIndexingMode is FieldIndexingMode.Exact
+            || binding.Analyzer is null)
         {
             return originalTerm;
         }
 
         var analyzer = binding.Analyzer!;
         analyzer.GetOutputBuffersSize(originalTerm.Length, out int outputSize, out int tokenSize);
-
+        Debug.Assert(outputSize < 1024 * 1024);
+        Debug.Assert(Unsafe.SizeOf<Token>() * tokenSize < 1024 * 1024);
```

Comment:
Please use 2nd parameter of Debug.Assert and pass a message. If it throws then it's easier to find it in the code.